src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -237,43 +194,23 @@ private async Task DoUpdate(string tempFile, int retry = 0)
             }
         }
 
-        private void EnsureOptions(AzureAppConfigurationOptions azconfigOptions)
+        private void CreateScopeToken(AzureAppConfigurationOptions azconfigOptions)
         {
             if (azconfigOptions == null)
             {
                 throw new ArgumentNullException(nameof(azconfigOptions));
             }
 
-            if (azconfigOptions.ConnectionString == null)
-            {
-                throw new InvalidOperationException("An Azure App Configuration connection string is required.");
-            }
-
-            OfflineFileCacheOptions options = _options ?? new OfflineFileCacheOptions();
-
-            if ((options.Key == null) || (options.SignKey == null) || (options.IV == null))
-            {
-                byte[] secret = Convert.FromBase64String(ConnectionStringParser.Parse(azconfigOptions.ConnectionString, "Secret"));
-                using (SHA256 sha256 = SHA256.Create())
-                {
-                    byte[] hash = sha256.ComputeHash(secret);
-
-                    options.Key = options.Key ?? hash;
-                    options.SignKey = options.SignKey ?? hash;
-                    options.IV = options.IV ?? hash.Take(16).ToArray();
-                }
-            }
-
             if (string.IsNullOrEmpty(_scopeToken))
             {
-                //
                 // The default scope token is the configuration store endpoint combined with all of the key-value filters
-
-                string endpoint = ConnectionStringParser.Parse(azconfigOptions.ConnectionString, "Endpoint");
-
+                string endpoint = (azconfigOptions.Endpoint != null)
+                                  ? azconfigOptions.Endpoint.ToString() 
+                                  : ConnectionStringParser.Parse(azconfigOptions.ConnectionString, "Endpoint");
+                
                 if (string.IsNullOrWhiteSpace(endpoint))

```

Comment:
How can this happen?

Reply:
That would be if someone passed a cooked AppConfigOptions that doesn't have correct endpoint or connection string? In that case wouldn't it be arg exception?

Reply:
I checked that this can never happen. If the connection string is invalid or endpoint is missing, AppConfig SDK will throw an exception during `AzureAppConfigurationSource.Build()` as it tries to create the `ConfigurationClient`.